lib/libzfs/libzfs_changelist.c

```diff
@@ -686,8 +697,8 @@ changelist_gather(zfs_handle_t *zhp, zfs_prop_t prop, int gather_flags,
 	else if (clp->cl_prop == ZFS_PROP_SHARESMB)
 		clp->cl_shareprop = ZFS_PROP_SHARENFS;
 
-	if (clp->cl_prop == ZFS_PROP_MOUNTPOINT &&
-	    (clp->cl_gflags & CL_GATHER_MOUNT_ALWAYS) == 0) {
+	if (clp->cl_prop == ZFS_PROP_MOUNTPOINT && (clp->cl_gflags &
+	    CL_GATHER_ITER_MOUNTED) == CL_GATHER_ITER_MOUNTED) {
```

Comment:
The additional comparison against `CL_GATHER_ITER_MOUNTED` looks to be redundant.